src/3d/framegraph/qgsframegraph.cpp

```diff
@@ -843,45 +641,24 @@ QString QgsFrameGraph::dumpFrameGraph() const
   while ( top->parent() && dynamic_cast<Qt3DRender::QFrameGraphNode *>( top->parent() ) )
     top = top->parent();
 
-  QgsFgUtils::FgDumpContext context;
+  QgsFrameGraphUtils::FgDumpContext context;
   context.lowestId = mMainCamera->id().id();
-  QStringList strList = QgsFgUtils::dumpFrameGraph( dynamic_cast<Qt3DRender::QFrameGraphNode *>( top ), context );
+  QStringList strList = QgsFrameGraphUtils::dumpFrameGraph( dynamic_cast<Qt3DRender::QFrameGraphNode *>( top ), context );
 
-  return qPrintable( strList.join( "\n" ) ) + QString( "\n" );
+  return strList.join( "\n" ) + QString( "\n" );
 }
 
 QString QgsFrameGraph::dumpSceneGraph() const
 {
-  QStringList strList = QgsFgUtils::dumpSceneGraph( mRootEntity, QgsFgUtils::FgDumpContext() );
-  return qPrintable( strList.join( "\n" ) ) + QString( "\n" );
+  QStringList strList = QgsFrameGraphUtils::dumpSceneGraph( mRootEntity, QgsFrameGraphUtils::FgDumpContext() );
+  return strList.join( "\n" ) + QString( "\n" );
 }
 
 void QgsFrameGraph::setClearColor( const QColor &clearColor )
 {
-  mForwardClearBuffers->setClearColor( clearColor );
-}
-
-void QgsFrameGraph::setShadowRenderingEnabled( bool enabled )
-{
-  mShadowRenderingEnabled = enabled;
-  mPostprocessingEntity->setShadowRenderingEnabled( mShadowRenderingEnabled );
-  if ( mShadowRenderingEnabled )
-    mShadowSceneEntitiesFilter->setEnabled( true );
-  else
-    mShadowSceneEntitiesFilter->setEnabled( false );
-}
-
-void QgsFrameGraph::setShadowBias( float shadowBias )
-{
-  mShadowBias = shadowBias;
-  mPostprocessingEntity->setShadowBias( mShadowBias );
-}
-
-void QgsFrameGraph::setShadowMapResolution( int resolution )
-{
-  mShadowMapResolution = resolution;
-  mShadowMapTexture->setWidth( mShadowMapResolution );
-  mShadowMapTexture->setHeight( mShadowMapResolution );
+  QgsForwardRenderView *rv = forwardRenderView();
+  if ( rv )
+    rv->setClearColor( clearColor );
```

Comment:
forward render must always exist, so let's do just `forwardRenderView()->setClearColor( clearColor );` like elsewhere.

There are couple more places below with needless check if you search for "if ( rv )"